idaes/core/util/model_diagnostics.py

```diff
@@ -1052,6 +1080,168 @@ def display_near_parallel_variables(self, stream=None):
     # TODO: Block triangularization analysis
     # Number and size of blocks, polynomial degree of 1x1 blocks, simple pivot check of moderate sized sub-blocks?
 
+    def _collect_constraint_mismatches(self, descend_into=True):
+        walker = ConstraintTermAnalysisVisitor(
+            term_mismatch_tolerance=self.config.constraint_term_mismatch_tolerance,
+            term_cancellation_tolerance=self.config.constraint_term_cancellation_tolerance,
+            term_zero_tolerance=self.config.constraint_term_zero_tolerance,
+        )
+
+        mismatch = []
+        cancellation = []
+        constant = []
+
+        for c in self._model.component_data_objects(
+            Constraint, descend_into=descend_into
+        ):
+            _, expr_mismatch, expr_cancellation, expr_constant = walker.walk_expression(
+                c.expr
+            )
+
+            if len(expr_mismatch) > 0:
+                mismatch.append(f"{c.name}: {len(expr_mismatch)} mismatched term(s)")
+
+            if len(expr_cancellation) > 0:
+                cancellation.append(
+                    f"{c.name}: {len(expr_cancellation)} potential cancelling term(s)"
+                )
+
+            if expr_constant:
+                constant.append(c.name)
+
+        return mismatch, cancellation, constant
+
+    def display_constraints_with_mismatched_terms(self, stream=None):
+        """
+        Display constraints in model which contain additive terms of significantly different magnitude.
+
+        Args:
+            stream: I/O object to write report to (default = stdout)
+
+        Returns:
+            None
+
+        """
+        if stream is None:
+            stream = sys.stdout
+
+        mismatch, _, _ = self._collect_constraint_mismatches()
+
+        # Write the output
+        _write_report_section(
+            stream=stream,
+            lines_list=mismatch,
+            title="The following constraints have mismatched terms:",
+            end_line="Call display_problematic_constraint_terms(constraint) for more information.",
+            header="=",
+            footer="=",
+        )
+
+    def display_constraints_with_canceling_terms(self, stream=None):
+        """
+        Display constraints in model which contain additive terms which potentially cancel each other.
+
```

Comment:
Worth mentioning that, while we've caught most common forms, some perfectly benign equality constraints will run afoul of this test when the constraint is satisfied.